agent/internal/containers.go

```diff
@@ -212,3 +232,168 @@ func containerEnvVars(cont cproto.Container) []string {
 		fmt.Sprintf("DET_SLOT_IDS=[%s]", strings.Join(slotIds, ",")),
 	}
 }
+
+func (c *containerManager) reattachContainers(
+	ctx *actor.Context, expectedSurvivors []aproto.ContainerReattach) (
+	[]aproto.ContainerReattachAck, error) {
+	result := make([]aproto.ContainerReattachAck, 0, len(expectedSurvivors))
+
+	runningContainers, err := c.listRunningContainers(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, survivor := range expectedSurvivors {
```

Comment:
nit, it's not necessarily a survivor, but an expectedSurvivor. i had to double take reading. maybe just actual/expected? idk